source/engine/graphics/system/ObjectSystem.cpp

```diff
@@ -17,6 +17,21 @@ bool IsViewedByFrustum(const nc::Frustum& frustum, float maxMeshExtent, DirectX:
     DirectX::XMStoreVector3(&sphere.center, transform.r[3]);
     return nc::physics::Intersect(frustum, sphere);
 }
+
+template<typename T>
+concept AnimatableComponent = std::same_as<T, nc::graphics::MeshRenderer> || std::same_as<T, nc::graphics::ToonRenderer>;
+
+template <AnimatableComponent T>
+uint32_t GetSkeletalAnimationIndex(const T* renderer, const nc::graphics::SkeletalAnimationSystemState& state)
+{
+    auto skeletalAnimIndex = UINT32_MAX;
+    auto iter = state.animationIndices.find(renderer->ParentEntity().Index());
+    if(iter != state.animationIndices.end())
```

Comment:
Personal opinion, but I find it more readable as:
```cpp
return iter != state.animationIndices.end() ? iter->second : UINT32_MAX;
```

Also, purely for academic purposes, you get guaranteed copy elision.